airflow/models/connection.py

```diff
@@ -43,6 +45,23 @@ def parse_netloc_to_hostname(*args, **kwargs):
     return _parse_netloc_to_hostname(*args, **kwargs)
 
 
+def sanitize_conn_id(conn_id: str | None) -> str | None:
```

Comment:
What is going the be the user experience on the UI side if a user enters a malformed connection ID? I am just worried that it will fail silently without telling the user why

Reply:
it's going to be the default behaviour in case a malformed one is encountered. Happy to raise an exception, if that's preferred but did not want to raise an exception in an unexpected location. Any suggestions?

Reply:
I can see some other places in this file where when it is not valid, returning `None` seems to be legit. LGTM then